### ISSUE TITLE
Crash when copying recipe, then on adding new steps in mash, ferment, boil, others

### ISSUE THREAD

**Issue by stripler675** [Brewtarget/brewtarget#868]
Compiled from git clone 29-Oct. Fedora 40. Had to comment out stacktrace_backtrace from CMakeLists.txt, because it was too much of a pain to get boost stacktrace_backtrace working.

Restarting after the crash, the step or item is present in the list. So if I'm adding a mash step, it crashes, when reopened the correct new mash step is there. Example crashes.
 
Copy recipe to new recipe (in this case the new recipe does not exist)
```
(brewtarget:61230): Gtk-WARNING **: 18:06:16.444: Failed to measure available space: The specified location is not supported
(brewtarget:61230): Gtk-WARNING **: 18:06:16.444: Failed to measure available space: The specified location is not supported
[18:07:24.756] (1dte18twcg) WARNING : bool {anonymous}::isUnusedInstanceOfUseOf(NE&) [with NE = Equipment] Equipment # 74 is unexpectedly already used in recipe # 64  [model/Recipe.cpp:129]
[18:07:24.758] (1dte18twcg) ERROR : ASSERT: "*this == copiedFrom" in file /localPath/brewtarget/src/model/NamedEntity.cpp, line 161  [model/NamedEntity.cpp:161]
Aborted (core dumped)
```

Adding new fermentation step
```
(brewtarget:61016): Gtk-WARNING **: 18:01:23.141: Failed to measure available space: The specified location is not supported
(brewtarget:61016): Gtk-WARNING **: 18:01:23.141: Failed to measure available space: The specified location is not supported
[18:02:16.503] (1dokf7rhmo) ERROR : ASSERT: "current + doSomething < this->derived().rows.size()" in file /localPath/brewtarget/src/tableModels/StepTableModelBase.h, line 156  [tableModels/StepTableModelBase.h:156]
Aborted (core dumped)
```

**Comment by matty0ung** [comment]
I'll take a look.  The first assert is a bug in the "copy recipe" functionality.  The second might well be a knock-on effect of the first.

**Comment by matty0ung** [comment]
I have a fix in the works for this.  It's a bit more than a one-liner, but should have a side benefit of eliminating some duplicated logic in the code.

**Comment by stripler675** [comment]
Thanks. Will check it.

**Comment by matty0ung** [comment]
Copy recipe is fixed in [release 4.0.9](https://github.com/Brewtarget/brewtarget/releases/tag/v4.0.9).  However, I'm still getting the crash on add mash step, so I'll have a look at that next.

**Comment by matty0ung** [comment]
Remaining fixes for this in the [4.0.10 release](https://github.com/Brewtarget/brewtarget/releases/tag/v4.0.10).
